src/main/java/org/opensearch/security/configuration/PrivilegesInterceptorImpl.java

```diff
@@ -95,9 +96,10 @@ private boolean isTenantAllowed(final ActionRequest request, final String action
      */
     @Override
     public ReplaceResult replaceDashboardsIndex(final ActionRequest request, final String action, final User user, final DynamicConfigModel config,
+                                                final TenancyConfigModel tenancyconfig,
                                                 final Resolved requestedResolved, final Map<String, Boolean> tenants) {
 
-        final boolean enabled = config.isDashboardsMultitenancyEnabled();//config.dynamic.kibana.multitenancy_enabled;
+        final boolean enabled = tenancyconfig.isDashboardsMultitenancyEnabled();
```

Comment:
Hi @abhivka7, thanks for this contribution. I see the point that you are trying to extend the original multi-tenancy feature with this change. So do we need a feature flag for this dynamically enable/disable multi-tenancy feature? Or it will be applied directly by enable multi tenancy?

Reply:
We don't meed a flag for these changes. We will keep multi-tenancy enabled for new domains and changes can be applied directly.